### PR TITLE
Setup initial project structure

### LINKED ISSUE #1: Setup shell/file structure
- [x] programming.interface() should be the same as the old project
- [x] should work in both the browser and node
- [x] mock return values of all functions

### PR DESCRIPTION
**Pull request by postatum**
Fixes #1 

Tests are already added here (without Travis).
Run with `npm run-script test-integration`.
Browser version is created with `npm run-script browserify`.

~In node example, lib should be installed using local version and not version specified in `examples/nodejs/package.json` (because it will install npm version of clj lib).~

To get rid of mocking, `src/mock.js` should be deleted and `src/canonocal.js & src/expanded.js` should be implemented.
Files in `test/fixtures` are used by mock but they should stay when mock is removed because they are used for integration tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "rewire": "^2.5.2",
+    "standard": "^10.0.2"
+  },
+  "dependencies": {
+    "lodash": "^4.17.4",
+    "yaml-js": "^0.1.3"
+  },
+  "files": [
+    "src/**",
+    "test/**"
```
Comment (jstoiko) [inline package.json]: no need to include the test/ folder in the npm package. let's add LICENSE though.

**Comment by jstoiko** [comment]
> In node example, lib should be installed using local version and not version specified in examples/nodejs/package.json (because it will install npm version of clj lib).

I took care of that

File: package.json (hunk 1 above)
Comment (postatum) [inline package.json]: there's a need for that as long as code uses mocks which are located in tests. without this package won't have code in tests and thus main code wont work as well

**Comment by postatum** [comment]
> I think it will make things easier to work with for the one working on this.

Not sure how this will make things easier but I'll do as you say.
Right now mock is a single line in expanded and canonical form code and mock.js. To get rid of mock - delete 2 lines and a file.
After this change mock will be huge chunks of data and funcs to work with that data in exchange and canonical form code.

**Comment by postatum** [comment]
Fixed all notes and added travis file

**Pull request merged**